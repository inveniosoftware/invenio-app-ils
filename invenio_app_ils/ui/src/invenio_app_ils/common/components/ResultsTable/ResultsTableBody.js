import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';

export default class ResultsTableBody extends Component {
  renderCell = (cell, column, id, colIndex) => {
    return (
      <Table.Cell key={colIndex + '-' + id} data-test={column + '-' + id}>
        {cell}
      </Table.Cell>
    );
  };

  renderRow = (columns, rows) => {
    return rows.map(row => {
      const withRowAction = this.props.rowActionClickHandler ? (
        <Button
          circular
          compact
          icon="eye"
          onClick={() => {
            this.props.rowActionClickHandler(row);
          }}
          data-test={'btn-view-details-' + row.ID}
        />
      ) : null;

      return (
        <Table.Row key={row.ID} data-test={row.ID}>
          <Table.Cell textAlign="center">{withRowAction}</Table.Cell>
          {columns.map((column, idx) =>
            this.renderCell(
              row[column] ? row[column] : '-',
              column,
              row.ID,
              idx
            )
          )}
        </Table.Row>
      );
    });
  };

  render() {
    const { columns, rows, detailsURL } = this.props;

    return <Table.Body>{this.renderRow(columns, rows, detailsURL)}</Table.Body>;
  }
}

ResultsTableBody.propTypes = {
  columns: PropTypes.array.isRequired,
  rowActionClickHandler: PropTypes.func,
};

ResultsTableBody.defaultProps = {
  rowActionClickHandler: null,
};
