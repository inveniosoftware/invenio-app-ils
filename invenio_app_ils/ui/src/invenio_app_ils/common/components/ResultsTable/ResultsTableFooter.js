import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import PropTypes from 'prop-types';

export default class ResultsTableFooter extends Component {
  constructor(props) {
    super(props);
    this.showAllButton = this.props.showAllButton;
    this.colspan = this.props.columnsNumber;
  }

  _renderFooter = () => {
    const itemNumber = this.props.allRowsNumber;
    return itemNumber > this.props.showMaxRows ? (
      <Table.Footer fullWidth data-test="footer">
        <Table.Row>
          <Table.HeaderCell colSpan={this.colspan + 1} textAlign="right">
            <span>
              Showing first {this.props.showMaxRows} entries of{' '}
              {this.props.allRowsNumber}{' '}
            </span>
            <span>{this.showAllButton}</span>
          </Table.HeaderCell>
        </Table.Row>
      </Table.Footer>
    ) : null;
  };

  render() {
    return this._renderFooter();
  }
}

ResultsTableFooter.propTypes = {
  allRowsNumber: PropTypes.number.isRequired,
  columnsNumber: PropTypes.number.isRequired,
  showMaxRows: PropTypes.number.isRequired,
  showAllButton: PropTypes.node,
};
