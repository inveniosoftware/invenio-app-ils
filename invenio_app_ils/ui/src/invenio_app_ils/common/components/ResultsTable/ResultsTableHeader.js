import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';

export default class ResultsTableHeader extends Component {
  _renderHeaderCell = cellName => {
    if (cellName === 'ID') {
      return (
        <Table.HeaderCell width={2} key={cellName}>
          {cellName}
        </Table.HeaderCell>
      );
    } else {
      return <Table.HeaderCell key={cellName}>{cellName}</Table.HeaderCell>;
    }
  };

  render() {
    const { columns, withRowAction } = this.props;
    const headerColumns = columns.map(column => this._renderHeaderCell(column));
    return (
      <Table.Header>
        <Table.Row data-test="header">
          <Table.HeaderCell width={withRowAction ? 2 : 1} collapsing />
          {headerColumns}
        </Table.Row>
      </Table.Header>
    );
  }
}

ResultsTableHeader.propTypes = {
  columns: PropTypes.array.isRequired,
  withRowAction: PropTypes.bool,
};

ResultsTableHeader.defaultProps = {
  withRowAction: false,
};
