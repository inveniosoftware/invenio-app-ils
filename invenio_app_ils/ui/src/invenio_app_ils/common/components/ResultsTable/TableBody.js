import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';

export default class TableBody extends Component {
  constructor(props) {
    super(props);
    this.detailsClickHandler = this.props.detailsClickHandler;
  }

  _renderCell = (cell, column, id) => {
    return <Table.Cell key={column + id}>{cell}</Table.Cell>;
  };

  _renderRow = (columns, rows) => {
    return rows.map(row => (
      <Table.Row key={row.ID} data-test={row.ID}>
        <Table.Cell key="details" textAlign="center">
          <Button
            circular
            compact
            icon="eye"
            onClick={() => {
              this.detailsClickHandler(row.ID);
            }}
          />
        </Table.Cell>
        {columns.map(column =>
          this._renderCell(row[column] ? row[column] : '-', column, row.ID)
        )}
      </Table.Row>
    ));
  };

  render() {
    const { columns, rows, detailsURL } = this.props;

    return (
      <Table.Body>{this._renderRow(columns, rows, detailsURL)}</Table.Body>
    );
  }
}

TableBody.propTypes = {
  columns: PropTypes.array.isRequired,
  detailsClickHandler: PropTypes.func.isRequired,
};
