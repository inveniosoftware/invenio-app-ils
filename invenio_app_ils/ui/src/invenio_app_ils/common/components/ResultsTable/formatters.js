import { fromISO, toShortDate, toShortDateTime } from '../../api/date';
import isEmpty from 'lodash/isEmpty';
import assign from 'lodash/assign';

function formatLoanToTableView(loan) {
  return {
    ID: loan.loan_pid ? loan.loan_pid : loan.id,
    Created: toShortDateTime(fromISO(loan.created)),
    Updated: toShortDateTime(fromISO(loan.updated)),
    'Patron ID': loan.metadata.patron_pid,
    'Document ID': loan.metadata.document_pid,
    State: loan.metadata.state,
    'Item barcode': loan.metadata.item ? loan.metadata.item.barcode : null,
    'Start date': toShortDate(fromISO(loan.metadata.start_date)),
    'End date': toShortDate(fromISO(loan.metadata.end_date)),
    'Transaction date': toShortDateTime(
      fromISO(loan.metadata.transaction_date)
    ),
    'Expiration date': toShortDateTime(
      fromISO(loan.metadata.request_expire_date)
    ),
    Renewals: loan.metadata.extension_count,
  };
}

function formatDocumentToTableView(document, seriesPid = null) {
  let serialized = {
    ID: document.document_pid ? document.document_pid : document.id,
    Created: toShortDate(fromISO(document.created)),
    Updated: toShortDate(fromISO(document.updated)),
  };
  if (!isEmpty(document.metadata)) {
    serialized['Title'] = document.metadata.title;
    serialized['Authors'] = document.metadata.authors;
    if (document.metadata.circulation) {
      serialized['Requests'] = document.metadata.circulation.pending_loans;
      serialized['Items'] = document.metadata.circulation.has_items_for_loan;
    }
    // Include volume number for series if it's given
    if (seriesPid) {
      const seriesObj = document.metadata.series_objs.find(
        obj => obj.pid === seriesPid
      );
      serialized['Volume'] = seriesObj ? seriesObj.volume : '';
    }
  }
  return serialized;
}

function formatItemToTableView(item) {
  return {
    ID: item.item_pid ? item.item_pid : item.id,
    Created: toShortDate(fromISO(item.created)),
    Updated: toShortDate(fromISO(item.updated)),
    Barcode: item.metadata.barcode,
    'Document ID': item.metadata.document_pid,
    Status: item.metadata.status,
    Medium: item.metadata.medium,
    'Circulation status': item.metadata.circulation_status
      ? item.metadata.circulation_status.state
      : null,
    'Internal location': item.metadata.internal_location
      ? item.metadata.internal_location.name
      : null,
    Location: item.metadata.internal_location.location
      ? item.metadata.internal_location.location.name
      : null,
    Shelf: item.metadata.shelf,
  };
}

function formatEItemToTableView(eitem) {
  return {
    ID: eitem.eitem_pid ? eitem.eitem_pid : eitem.id,
    'Document ID': eitem.metadata.document_pid,
    'Open access': eitem.metadata.open_access ? 'Yes' : 'No',
    Created: toShortDate(fromISO(eitem.created)),
    Updated: toShortDate(fromISO(eitem.updated)),
    Description: eitem.metadata.description,
    'Internal Notes': eitem.metadata.internal_notes,
  };
}

function formatLocationToTableView(location) {
  return {
    ID: location.location_pid ? location.location_pid : location.id,
    Created: toShortDate(fromISO(location.created)),
    Updated: toShortDate(fromISO(location.updated)),
    Link: location.links.self,
    Name: location.metadata.name,
    Address: location.metadata.address,
    Email: location.metadata.email,
  };
}

function formatInternalLocationToTableView(internalLoc) {
  let entry = {
    ID: internalLoc.item_pid ? internalLoc.item_pid : internalLoc.id,
    Created: toShortDate(fromISO(internalLoc.created)),
    Updated: toShortDate(fromISO(internalLoc.updated)),
    Link: internalLoc.links.self,
  };
  if (!isEmpty(internalLoc.metadata)) {
    assign(entry, {
      Name: internalLoc.metadata.name,
      'Physical location': internalLoc.metadata.physical_location,
      'Location e-mail': internalLoc.metadata.location.email,
      'Location name': internalLoc.metadata.location.name,
      'Location id': internalLoc.metadata.location.location_pid,
    });
  }
  return entry;
}

function formatSeriesToTableView(series) {
  let serialized = {
    ID: series.series_pid ? series.series_pid : series.id,
    Created: toShortDate(fromISO(series.created)),
    Updated: toShortDate(fromISO(series.updated)),
  };
  if (!isEmpty(series.metadata)) {
    serialized['Title'] = series.metadata.title;
    serialized['Authors'] = series.metadata.authors;
    serialized['Mode of Issuance'] = series.metadata.mode_of_issuance;
  }
  return serialized;
}

function formatPatronToTableView(patron) {
  return {
    ID: patron.metadata.id,
    Name: patron.metadata.name,
    Email: patron.metadata.email,
  };
}

function formatPidTypeToName(pidType) {
  switch (pidType) {
    case 'docid':
      return 'Document';
    case 'serid':
      return 'Series';
    default:
      console.warn(`Unknown pid type: ${pidType}`);
  }
}

function formatRelatedToTableView(related, relation) {
  return {
    ID: related.pid,
    Type: formatPidTypeToName(related.pid_type),
    Title: related.title,
    Edition: related.edition,
    Language: related.language,
    Relation: relation.label,
  };
}

export const formatter = {
  loan: { toTable: formatLoanToTableView },
  document: { toTable: formatDocumentToTableView },
  item: { toTable: formatItemToTableView },
  eitem: { toTable: formatEItemToTableView },
  location: { toTable: formatLocationToTableView },
  internalLocation: { toTable: formatInternalLocationToTableView },
  series: { toTable: formatSeriesToTableView },
  patron: { toTable: formatPatronToTableView },
  related: { toTable: formatRelatedToTableView },
};
