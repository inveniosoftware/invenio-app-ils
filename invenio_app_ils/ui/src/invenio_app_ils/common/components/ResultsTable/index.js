export { ResultsTable } from './ResultsTable';
export { formatter as ResultsTableFormatter } from './formatters';
