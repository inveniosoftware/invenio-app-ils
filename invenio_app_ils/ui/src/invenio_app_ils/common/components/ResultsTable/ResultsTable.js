import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Segment, Message, Header, Table, Popup } from 'semantic-ui-react';
import ResultsTableHeader from './ResultsTableHeader';
import ResultsTableBody from './ResultsTableBody';
import ResultsTableFooter from './ResultsTableFooter';

export class ResultsTable extends Component {
  constructor(props) {
    super(props);
    this.actionClickHandler = this.props.actionClickHandler;
    this.showAllButton = this.props.showAllButton;
    this.actionComponent = this.props.actionComponent;
  }

  _renderTable = () => {
    const { rows, showMaxRows, singleLine, fixed } = this.props;
    const columns = rows ? Object.keys(rows[0]) : [];
    return (
      <Table
        striped
        {...(singleLine ? { singleLine: true } : {})}
        {...(fixed ? { fixed: true } : {})}
        selectable
      >
        <ResultsTableHeader columns={columns} />
        <ResultsTableBody
          columns={columns}
          rows={rows.slice(0, showMaxRows)}
          actionClickHandler={this.actionClickHandler}
          actionComponent={this.actionComponent}
        />
        <ResultsTableFooter
          allRowsNumber={rows.length}
          showMaxRows={this.props.showMaxRows}
          showAllButton={this.showAllButton}
          columnsNumber={columns.length}
        />
      </Table>
    );
  };

  _renderResultsOrEmpty() {
    const { rows, name } = this.props;
    return rows.length ? (
      this._renderTable(rows)
    ) : (
      <Message data-test="no-results">There are no {name}</Message>
    );
  }

  render() {
    const { name, popup } = this.props;
    if (popup) {
      const header = <Header as="h3">{name}</Header>;
      return (
        <Segment>
          <Popup content={popup} trigger={header} />
          {this._renderResultsOrEmpty()}
        </Segment>
      );
    }
    return (
      <Segment>
        <Header as="h3">{name}</Header>
        {this._renderResultsOrEmpty()}
      </Segment>
    );
  }
}

ResultsTable.propTypes = {
  rows: PropTypes.array.isRequired,
  showMaxRows: PropTypes.number,
  name: PropTypes.string,
  actionClickHandler: PropTypes.func.isRequired,
  showAllButton: PropTypes.node,
  actionComponent: PropTypes.node,
  wrapped: PropTypes.bool,
  popup: PropTypes.string,
};

ResultsTable.defaultProps = {
  showMaxRows: 10,
  name: 'results',
  actionComponent: null,
};
