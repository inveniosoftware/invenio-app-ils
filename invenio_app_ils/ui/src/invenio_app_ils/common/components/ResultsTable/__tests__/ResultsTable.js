import { shallow, mount } from 'enzyme/build';
import React from 'react';
import { ResultsTable } from '../ResultsTable';
import { Settings } from 'luxon';
import { loan as loanApi } from '../../../api';
import { Button } from 'semantic-ui-react';

Settings.defaultZoneName = 'utc';

const d = '2018-01-01';

describe('ResultsTable tests', () => {
  let component;

  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  it('should load the ResultTable component', () => {
    const component = shallow(
      <ResultsTable
        history={() => {}}
        rows={[]}
        actionClickHandler={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the show all button when showing only a few rows', () => {
    const data = [
      {
        loan_pid: 'loan1',
        ID: '1',
        patron_pid: 'patron_1',
        updated: d,
        start_date: d,
        end_date: d,
      },
      {
        loan_pid: 'loan2',
        ID: '2',
        patron_pid: 'patron_2',
        updated: d,
        start_date: d,
        end_date: d,
      },
    ];

    const button = () => {
      return (
        <Button size="small" onClick={() => {}}>
          Show all
        </Button>
      );
    };

    component = mount(
      <ResultsTable
        history={() => {}}
        rows={data}
        showAllButton={button()}
        actionClickHandler={() => {}}
        showMaxRows={1}
      />
    );

    expect(component).toMatchSnapshot();
    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    const showAll = footer.find('button');
    expect(footer).toHaveLength(1);
    expect(showAll).toHaveLength(1);
  });

  it('should not render the show all button when showing only a few rows', () => {
    const data = [
      {
        ID: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        start_date: d,
        end_date: d,
      },
      {
        ID: 'loan2',
        patron_pid: 'patron_2',
        updated: d,
        start_date: d,
        end_date: d,
      },
    ];

    component = mount(
      <ResultsTable
        history={() => {}}
        rows={data}
        actionClickHandler={() => {}}
        showMaxRows={3}
      />
    );

    expect(component).toMatchSnapshot();
    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should call click handler on view details click', () => {
    const mockedClickHandler = jest.fn();

    const results = [
      {
        ID: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        start_date: d,
        end_date: d,
      },
      {
        ID: 'loan2',
        patron_pid: 'patron_2',
        updated: d,
        start_date: d,
        end_date: d,
      },
    ];

    component = mount(
      <ResultsTable rows={results} actionClickHandler={mockedClickHandler} />
    );
    const firstId = results[0].ID;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('button');
    button.simulate('click');
    expect(mockedClickHandler).toHaveBeenCalledWith(firstId);
  });

  it('should call show all click handler on show all click', () => {
    const mockedClickHandler = jest.fn();
    const mockedHistoryPush = jest.fn();
    const historyFn = {
      push: mockedHistoryPush,
    };

    const results = [
      {
        ID: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        start_date: d,
        end_date: d,
      },
      {
        ID: 'loan2',
        patron_pid: 'patron_2',
        updated: d,
        start_date: d,
        end_date: d,
      },
    ];

    const buttonObj = () => {
      return (
        <Button
          size="small"
          onClick={() => {
            mockedHistoryPush();
          }}
        >
          Show all
        </Button>
      );
    };

    component = mount(
      <ResultsTable
        rows={results}
        showMaxRows={1}
        history={historyFn}
        showAllButton={buttonObj()}
        actionClickHandler={() => {}}
      />
    );

    const button = component.find('TableFooter').find('button');
    button.simulate('click');
    expect(mockedHistoryPush).toHaveBeenCalled();
  });
});
