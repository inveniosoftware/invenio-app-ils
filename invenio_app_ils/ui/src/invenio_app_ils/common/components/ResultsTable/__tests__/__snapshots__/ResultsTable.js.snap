// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsTable tests should load the ResultTable component 1`] = `
<Segment>
  <Header
    as="h3"
  >
    results
  </Header>
  <Message
    data-test="no-results"
  >
    There are no 
    results
  </Message>
</Segment>
`;

exports[`ResultsTable tests should not render the show all button when showing only a few rows 1`] = `
<ResultsTable
  actionClickHandler={[Function]}
  actionComponent={null}
  history={[Function]}
  name="results"
  rows={
    Array [
      Object {
        "ID": "loan1",
        "end_date": "2018-01-01",
        "patron_pid": "patron_1",
        "start_date": "2018-01-01",
        "updated": "2018-01-01",
      },
      Object {
        "ID": "loan2",
        "end_date": "2018-01-01",
        "patron_pid": "patron_2",
        "start_date": "2018-01-01",
        "updated": "2018-01-01",
      },
    ]
  }
  showMaxRows={3}
>
  <Segment>
    <div
      className="ui segment"
    >
      <Header
        as="h3"
      >
        <h3
          className="ui header"
        >
          results
        </h3>
      </Header>
      <Table
        as="table"
        selectable={true}
        striped={true}
      >
        <table
          className="ui selectable striped table"
        >
          <ResultsTableHeader
            columns={
              Array [
                "ID",
                "patron_pid",
                "updated",
                "start_date",
                "end_date",
              ]
            }
          >
            <TableHeader
              as="thead"
            >
              <thead
                className=""
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                  data-test="header"
                >
                  <tr
                    className=""
                    data-test="header"
                  >
                    <TableHeaderCell
                      as="th"
                      collapsing={true}
                      width={2}
                    >
                      <TableCell
                        as="th"
                        className=""
                        collapsing={true}
                        width={2}
                      >
                        <th
                          className="collapsing two wide"
                        />
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="ID"
                      width={2}
                    >
                      <TableCell
                        as="th"
                        className=""
                        width={2}
                      >
                        <th
                          className="two wide"
                        >
                          ID
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="patron_pid"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          patron_pid
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="updated"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          updated
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="start_date"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          start_date
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="end_date"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          end_date
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                  </tr>
                </TableRow>
              </thead>
            </TableHeader>
          </ResultsTableHeader>
          <ResultsTableBody
            actionClickHandler={[Function]}
            actionComponent={null}
            columns={
              Array [
                "ID",
                "patron_pid",
                "updated",
                "start_date",
                "end_date",
              ]
            }
            rows={
              Array [
                Object {
                  "ID": "loan1",
                  "end_date": "2018-01-01",
                  "patron_pid": "patron_1",
                  "start_date": "2018-01-01",
                  "updated": "2018-01-01",
                },
                Object {
                  "ID": "loan2",
                  "end_date": "2018-01-01",
                  "patron_pid": "patron_2",
                  "start_date": "2018-01-01",
                  "updated": "2018-01-01",
                },
              ]
            }
          >
            <TableBody
              as="tbody"
            >
              <tbody
                className=""
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                  data-test="loan1"
                  key="loan1"
                >
                  <tr
                    className=""
                    data-test="loan1"
                  >
                    <TableCell
                      as="td"
                      key="details"
                      textAlign="center"
                    >
                      <td
                        className="center aligned"
                      >
                        <Button
                          as="button"
                          circular={true}
                          compact={true}
                          icon="eye"
                          onClick={[Function]}
                        >
                          <button
                            className="ui circular compact icon button"
                            onClick={[Function]}
                          >
                            <Icon
                              as="i"
                              name="eye"
                            >
                              <i
                                aria-hidden="true"
                                className="eye icon"
                              />
                            </Icon>
                          </button>
                        </Button>
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="ID-loan1"
                      key="0-loan1"
                    >
                      <td
                        className=""
                        data-test="ID-loan1"
                      >
                        loan1
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="patron_pid-loan1"
                      key="1-loan1"
                    >
                      <td
                        className=""
                        data-test="patron_pid-loan1"
                      >
                        patron_1
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="updated-loan1"
                      key="2-loan1"
                    >
                      <td
                        className=""
                        data-test="updated-loan1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="start_date-loan1"
                      key="3-loan1"
                    >
                      <td
                        className=""
                        data-test="start_date-loan1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="end_date-loan1"
                      key="4-loan1"
                    >
                      <td
                        className=""
                        data-test="end_date-loan1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                  </tr>
                </TableRow>
                <TableRow
                  as="tr"
                  cellAs="td"
                  data-test="loan2"
                  key="loan2"
                >
                  <tr
                    className=""
                    data-test="loan2"
                  >
                    <TableCell
                      as="td"
                      key="details"
                      textAlign="center"
                    >
                      <td
                        className="center aligned"
                      >
                        <Button
                          as="button"
                          circular={true}
                          compact={true}
                          icon="eye"
                          onClick={[Function]}
                        >
                          <button
                            className="ui circular compact icon button"
                            onClick={[Function]}
                          >
                            <Icon
                              as="i"
                              name="eye"
                            >
                              <i
                                aria-hidden="true"
                                className="eye icon"
                              />
                            </Icon>
                          </button>
                        </Button>
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="ID-loan2"
                      key="0-loan2"
                    >
                      <td
                        className=""
                        data-test="ID-loan2"
                      >
                        loan2
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="patron_pid-loan2"
                      key="1-loan2"
                    >
                      <td
                        className=""
                        data-test="patron_pid-loan2"
                      >
                        patron_2
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="updated-loan2"
                      key="2-loan2"
                    >
                      <td
                        className=""
                        data-test="updated-loan2"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="start_date-loan2"
                      key="3-loan2"
                    >
                      <td
                        className=""
                        data-test="start_date-loan2"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="end_date-loan2"
                      key="4-loan2"
                    >
                      <td
                        className=""
                        data-test="end_date-loan2"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                  </tr>
                </TableRow>
              </tbody>
            </TableBody>
          </ResultsTableBody>
          <ResultsTableFooter
            allRowsNumber={2}
            columnsNumber={5}
            showMaxRows={3}
          />
        </table>
      </Table>
    </div>
  </Segment>
</ResultsTable>
`;

exports[`ResultsTable tests should render the show all button when showing only a few rows 1`] = `
<ResultsTable
  actionClickHandler={[Function]}
  actionComponent={null}
  history={[Function]}
  name="results"
  rows={
    Array [
      Object {
        "ID": "1",
        "end_date": "2018-01-01",
        "loan_pid": "loan1",
        "patron_pid": "patron_1",
        "start_date": "2018-01-01",
        "updated": "2018-01-01",
      },
      Object {
        "ID": "2",
        "end_date": "2018-01-01",
        "loan_pid": "loan2",
        "patron_pid": "patron_2",
        "start_date": "2018-01-01",
        "updated": "2018-01-01",
      },
    ]
  }
  showAllButton={
    <Button
      as="button"
      onClick={[Function]}
      size="small"
    >
      Show all
    </Button>
  }
  showMaxRows={1}
>
  <Segment>
    <div
      className="ui segment"
    >
      <Header
        as="h3"
      >
        <h3
          className="ui header"
        >
          results
        </h3>
      </Header>
      <Table
        as="table"
        selectable={true}
        striped={true}
      >
        <table
          className="ui selectable striped table"
        >
          <ResultsTableHeader
            columns={
              Array [
                "loan_pid",
                "ID",
                "patron_pid",
                "updated",
                "start_date",
                "end_date",
              ]
            }
          >
            <TableHeader
              as="thead"
            >
              <thead
                className=""
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                  data-test="header"
                >
                  <tr
                    className=""
                    data-test="header"
                  >
                    <TableHeaderCell
                      as="th"
                      collapsing={true}
                      width={2}
                    >
                      <TableCell
                        as="th"
                        className=""
                        collapsing={true}
                        width={2}
                      >
                        <th
                          className="collapsing two wide"
                        />
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="loan_pid"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          loan_pid
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="ID"
                      width={2}
                    >
                      <TableCell
                        as="th"
                        className=""
                        width={2}
                      >
                        <th
                          className="two wide"
                        >
                          ID
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="patron_pid"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          patron_pid
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="updated"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          updated
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="start_date"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          start_date
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                    <TableHeaderCell
                      as="th"
                      key="end_date"
                    >
                      <TableCell
                        as="th"
                        className=""
                      >
                        <th
                          className=""
                        >
                          end_date
                        </th>
                      </TableCell>
                    </TableHeaderCell>
                  </tr>
                </TableRow>
              </thead>
            </TableHeader>
          </ResultsTableHeader>
          <ResultsTableBody
            actionClickHandler={[Function]}
            actionComponent={null}
            columns={
              Array [
                "loan_pid",
                "ID",
                "patron_pid",
                "updated",
                "start_date",
                "end_date",
              ]
            }
            rows={
              Array [
                Object {
                  "ID": "1",
                  "end_date": "2018-01-01",
                  "loan_pid": "loan1",
                  "patron_pid": "patron_1",
                  "start_date": "2018-01-01",
                  "updated": "2018-01-01",
                },
              ]
            }
          >
            <TableBody
              as="tbody"
            >
              <tbody
                className=""
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                  data-test="1"
                  key="1"
                >
                  <tr
                    className=""
                    data-test="1"
                  >
                    <TableCell
                      as="td"
                      key="details"
                      textAlign="center"
                    >
                      <td
                        className="center aligned"
                      >
                        <Button
                          as="button"
                          circular={true}
                          compact={true}
                          icon="eye"
                          onClick={[Function]}
                        >
                          <button
                            className="ui circular compact icon button"
                            onClick={[Function]}
                          >
                            <Icon
                              as="i"
                              name="eye"
                            >
                              <i
                                aria-hidden="true"
                                className="eye icon"
                              />
                            </Icon>
                          </button>
                        </Button>
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="loan_pid-1"
                      key="0-1"
                    >
                      <td
                        className=""
                        data-test="loan_pid-1"
                      >
                        loan1
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="ID-1"
                      key="1-1"
                    >
                      <td
                        className=""
                        data-test="ID-1"
                      >
                        1
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="patron_pid-1"
                      key="2-1"
                    >
                      <td
                        className=""
                        data-test="patron_pid-1"
                      >
                        patron_1
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="updated-1"
                      key="3-1"
                    >
                      <td
                        className=""
                        data-test="updated-1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="start_date-1"
                      key="4-1"
                    >
                      <td
                        className=""
                        data-test="start_date-1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                    <TableCell
                      as="td"
                      data-test="end_date-1"
                      key="5-1"
                    >
                      <td
                        className=""
                        data-test="end_date-1"
                      >
                        2018-01-01
                      </td>
                    </TableCell>
                  </tr>
                </TableRow>
              </tbody>
            </TableBody>
          </ResultsTableBody>
          <ResultsTableFooter
            allRowsNumber={2}
            columnsNumber={6}
            showAllButton={
              <Button
                as="button"
                onClick={[Function]}
                size="small"
              >
                Show all
              </Button>
            }
            showMaxRows={1}
          >
            <TableFooter
              as="tfoot"
              data-test="footer"
              fullWidth={true}
            >
              <TableHeader
                as="tfoot"
                data-test="footer"
                fullWidth={true}
              >
                <tfoot
                  className="full-width"
                  data-test="footer"
                >
                  <TableRow
                    as="tr"
                    cellAs="td"
                  >
                    <tr
                      className=""
                    >
                      <TableHeaderCell
                        as="th"
                        colSpan={7}
                        textAlign="right"
                      >
                        <TableCell
                          as="th"
                          className=""
                          colSpan={7}
                          textAlign="right"
                        >
                          <th
                            className="right aligned"
                            colSpan={7}
                          >
                            <span>
                              Showing first 
                              1
                               entries of
                               
                              2
                               
                            </span>
                            <span>
                              <Button
                                as="button"
                                onClick={[Function]}
                                size="small"
                              >
                                <button
                                  className="ui small button"
                                  onClick={[Function]}
                                >
                                  Show all
                                </button>
                              </Button>
                            </span>
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                    </tr>
                  </TableRow>
                </tfoot>
              </TableHeader>
            </TableFooter>
          </ResultsTableFooter>
        </table>
      </Table>
    </div>
  </Segment>
</ResultsTable>
`;
